pkg/ingress/controller/controller.go

```diff
@@ -824,10 +824,17 @@ func (c *Controller) ensureIngress(ing *nwv1.Ingress) error {
 			var policyRules []l7policies.CreateRuleOpts
 
 			if host != "" {
+
+				// Escape dot to be regex proof
+				hostRegex := strings.ReplaceAll(host, ".", "\\.")
+
+				// Handle wildcard
+				hostRegex = strings.ReplaceAll(hostRegex, "*", ".*")
```

Comment:
I haven't looked at the full context, but at first glance this doesn't seem to do what it's supposed to. I'm assuming (again: sorry I haven't actually checked properly!) that the user writes `*.example.com`. They expect that to match `foo.example.com`, but do they expect it to match `foo.bar.example.com`? I suspect not, but this looks like it would match it.

Reply:
Fair point I'll see what I can do about that.